docs/source/public_api/testproblems.rst

```diff
@@ -0,0 +1,24 @@
+probnum.testproblems
+========================
+
+Test problems for probabilistic linear solvers.
```

Comment:
Do you have a strong opinion on the naming? I am not sure, whether `testproblems` communicates the right thing. We do not only use it for tests, but because it is convenient to collect them somewhere and then I think we will use them everywhere. How about, alternatively, something like `problemzoo` (which is inspired by pytorch's modelzoo).

Reply:
I don't. I changed it to `problem_zoo`, but I'm open to other suggestions.